### DISCUSSION TITLE
Error: Chain 'eip155:420' not approved. Please use one of the following: eip155:5

### DISCUSSION THREAD

**Discussion by compojoom** [WalletConnect/walletconnect-monorepo#2774] [Developer Support]
We are trying to migrate to wallet connect v2 in this PR over here:
https://github.com/beamer-bridge/beamer/pull/1972

I'm trying to connect Metamask with the following chainIds:
[ 5, 420, 1442, 84531, 421613 ]

We are using EthereumProvider: https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/providers/ethereum-provider/src/EthereumProvider.ts

Initially I was passing the whole chains array above to the `chains` parameter of the initialize funcition. Metamask was greeting me with "Unsupported chain" exception. I can understand this as none of the chains except 5 is actually configured in Metamask. So I thought that I'll pass just 5 as required chain and everything else as optional. Now I'm able to connect, but whenever I try to switch chain I'm getting:

```
Error: Chain 'eip155:420' not approved. Please use one of the following: eip155:5
```

In the desktop Metamask extension if the chain is not configured, Metamask would offer to add the chain to the list of chains. 

The `rpcMap` that we pass already has an RPC url for that chain. 

How are we to convince a walletconnect connection to happen if we have a chain that is not configured in the wallet?

**Comment by crypblizz8** [comment]
Hey @compojoom, would you be able to reproduce with these wallets to help us find a better path to help you:
- [Web3Wallet](https://react-web3wallet.vercel.app/) (Recommend opening it on your mobile phone browser)
- [AlphaWallet](https://alphawallet.com/)
- [Zerion](https://zerion.io/)

**Comment by compojoom** [comment]
@crypblizz8 - hi, thanks for the reply!
Yes, I'm able to reproduce this issue with the Web3Wallet. 

Again, here is how I initialize the Provider:
```
  const provider = await WalletConnect.init({
    chains: [chains[0]],
    optionalChains: chains,
    projectId: BEAMER_PROJECT_ID,
    rpcMap: rpcList,
    showQrModal: true,
  });
```

the chains array consist of: [ 1, 10, 1101, 42161 ]

Whenever I try to change the network through the app I get:
```
Error: Chain 'eip155:10' not approved. Please use one of the following: eip155:1
```

I also did some more digging in Metamask and I can say the following. In metamask I have 1, 10 and 42161 configured. 

If I start my session with let's say 42161 set I can connect. Then if I try to switch the network and select mainnet (1, which has been provided to the chains array) everything seems fine in the beamer app, but the selected network in Metamask doesn't change.

If I try to select network 10, I get 
```
Error: Chain 'eip155:10' not approved. Please use one of the following: eip155:1
```

If I switch the network though Metamask - the beamer app reacts and the page refreshes. Now If I try to switch between 42161 and 10 - I no longer get the chain not approved error, but the chain doesn't actually change in Metamask.

**Comment by compojoom** [comment]
I hacked a bit on the walletconnect examples and I think I managed to reproduce it over here:
https://github.com/compojoom/web-examples/tree/bug

I added some wallet_switchEthereumChain calls:
https://github.com/compojoom/web-examples/blob/bug/dapps/react-dapp-v2-with-ethers/src/pages/index.tsx#L249;251

And I exchanged the UniversalProvider for the EthereumProvider:
https://github.com/compojoom/web-examples/blob/bug/dapps/react-dapp-v2-with-ethers/src/contexts/ClientContext.tsx#L153

If you start the app and then connect to mainnet, after that try to connect to chain 10 or 42161 you would get
```
 Error: Chain 'eip155:10' not approved. Please use one of the following: eip155:1,eip155:42161
```

It doesn't matter if I add 10 and 42161 to the optionalChains - the result is the same when I try to switch.  

Also - note the chain you have selected in Metamask when connecting. If you are with 42161 (arbitrum), then when you try to switch to 1 - it will work (because 1 is the "main chain"), but it will also work, when trying to go to 42161 - as this was the chain when connecting. But if you try to switch to 10 - you get the above error.

What are we doing wrong here?

**Comment by compojoom** [comment]
This particular problem has been resolved with 2.8.4
